### PR TITLE
feat(mls): added FetchGroupStateActionHandler FS-958

### PR DESCRIPTION
**Pull request by ghuMustafa**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-958" title="FS-958" target="_blank">TaskFS-958</a>  [iOS] 5.1.1.1 Fetch public group state
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- Client who wants to join an MLS group via an external commit we need to fetch the group state from the backend.
- Created a new `FetchGroupStateActionHandler`
- Added `FetchGroupStateActionHandler` action handler to `MLSRequestStrategy`

Needs releases with:

- https://github.com/wireapp/wire-ios-data-model/pull/1423

### Testing

- Added tests for `FetchGroupStateActionHandler`

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    override func handleResponse(_ response: ZMTransportResponse, action: FetchPublicGroupStateAction) {
+        var action = action
+
+        switch (response.httpStatus, response.payloadLabel()) {
+        case (200, _):
+            guard let data = response.rawData,
+                  let payload = try? JSONDecoder().decode(ResponsePayload.self, from: data)
+            else {
+                action.notifyResult(.failure(.malformedResponse))
```
Comment (David-Henner) [inline Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift:58]:
```suggestion
                action.fail(with: .malformedResponse)
```

Comment (ghuMustafa) [reply Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift]: changed.

File: Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        guard
+            !action.domain.isEmpty,
+            !action.conversationId.isEmpty
+        else {
+            action.fail(with: .emptyParameters)
+            return nil
+        }
+
+        return ZMTransportRequest(
+            path: "/conversations/\(action.domain)/\(action.conversationId)/groupinfo",
```
Comment (johnxnguyen) [inline Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift:43]: As I mentioned in the other PR, I think we should have the conversation ID as a UUID here and map it to a string when building the request.

Comment (ghuMustafa) [reply Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift]: Yes I have conversationId as UUID

File: Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        guard
+            !action.domain.isEmpty,
+            !action.conversationId.uuidString.isEmpty
+        else {
+            action.fail(with: .emptyParameters)
+            return nil
+        }
+
+        return ZMTransportRequest(
+            path: "/conversations/\(action.domain)/\(action.conversationId.uuidString)/groupinfo",
```
Comment (johnxnguyen) [inline Sources/Request Strategies/MLS/Actions/FetchPublicGroupStateActionHandler.swift:43]: Actually I think we need to make this `action.conversationId.transportString()`

**Review by johnxnguyen — CHANGES_REQUESTED** [review]

**Review by David-Henner — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

File: Cartfile.resolved [hunk 4]
```diff
@@ -6,7 +6,7 @@ github "wireapp/libsodium" "1.0.14"
 github "wireapp/ocmock" "v3.4.3_xcframework"
 github "wireapp/swift-protobuf" "1.15.0_xcframework"
 github "wireapp/wire-ios-cryptobox" "34.0.0"
-github "wireapp/wire-ios-data-model" "feat/mls"
+github "wireapp/wire-ios-data-model" "feat/mls-fetch-public-group-state"
```
Comment (David-Henner) [inline Cartfile.resolved:9]:
```suggestion
github "wireapp/wire-ios-data-model" "feat/mls"
```

**Pull request merged**